service/src/test/java/bio/terra/catalog/api/DatasetApiControllerTest.java

```diff
@@ -31,27 +38,80 @@
 @WebMvcTest
 class DatasetApiControllerTest {
 
+  private static final String API = "/api/v1/datasets";
+  private static final String API_ID = API + "/{id}";
```

Comment:
In theory it's possible to extract the values from the `RequestMapping` annotation. I don't think I've ever seen a spring test that does this though. I could take a look at it.

Reply:
It looks like this works:
```
var apiPath = DatasetsApi.class.getMethod("listDatasets").getAnnotation(RequestMapping.class).value()[0]
```
generates the string `/api/v1/datasets`. So this swaps a dependency on the path with a dependency on the openapi generated class (`DatasetsApi`) and the method name in that class `listDatasets`. Not sure if this is better or not; maybe duplicating the path is a feature so we know if we changed the path by accident?

Reply:
I guess the string approach is the less of two evils.

Reply:
Yeah I implemented this just to see how it would work out, and since the method lookup uses reflection you end up having to write out the entire signature to get the annotations. It's not great.
```
  private String apiPath(String apiName, Class<?>... parameterTypes) throws NoSuchMethodException {
    return DatasetsApi.class
        .getMethod(apiName, parameterTypes)
        .getAnnotation(RequestMapping.class)
        .value()[0];
  }

  @Test
  void deleteDataset() throws Exception {
    var datasetId = new DatasetId(UUID.randomUUID());
    mockMvc
        .perform(delete(apiPath("deleteDataset", UUID.class), datasetId.uuid()))
        .andExpect(status().is2xxSuccessful());
    verify(datasetService).deleteMetadata(user, datasetId);
  }
...